java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -728,13 +731,17 @@ private void addSerializer(Class<?> type, Serializer<?> serializer) {
       }
       classInfo = classInfoMap.get(type);
     }
+
     if (classInfo == null || typeTag != null || classId != classInfo.classId) {
-      classInfo = new ClassInfo(this, type, typeTag, serializer, classId);
-    } else {
-      classInfo.serializer = serializer;
+      classInfo = new ClassInfo(this, type, typeTag, null, classId);
```

Comment:
Perhaps when `typeTag != null`, we do not need to re-create `ClassInfo`, we can just reassign the `typeTagBytes` in `ClassInfo`, which can save the time of creating `ClassInfo` object.